src_graphql/github_domain.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects for github entities."""
+
+from __future__ import annotations
+
+import datetime
+
+from typing import Any, Dict, List, Optional, Type
+
+DEFAULT_TIMESTAMP: datetime.datetime = (
+    datetime.datetime(datetime.MINYEAR, 1, 1, tzinfo=datetime.timezone.utc))
+
+
+class Assignee:
+    """A class representing a assignee of a pull request."""
+
+    def __init__(
+        self,
+        name: str,
+        timestamp: datetime.datetime =DEFAULT_TIMESTAMP
+    ) -> None:
+        self.name = name
+        self.timestamp = timestamp
+
+    def set_timestamp(self, timestamp: datetime.datetime) -> None:
+        """Sets timestamp to assignee."""
+        self.timestamp = timestamp
+
+    def get_readable_waiting_time(self) -> str:
+        """"Returns redable wating time on review."""
+        delta = datetime.datetime.now(datetime.timezone.utc) - self.timestamp
+        days = delta.days
+        hours, _ = divmod(delta.seconds, 3600)
+        waiting_time = []
+        if days:
+            waiting_time.append(f'{days} day{1}'.format(
+                's' if days > 1 else ''))
+
+        if hours:
+            waiting_time.append(f'{hours} hour{1}'.format(
+                's' if hours > 1 else ''))
+
+        return ', '.join(waiting_time)
+
+    def __repr__(self) -> str:
+        return f'@{self.name} assigned on {self.timestamp}'
+
+
+class PullRequest:
+    """A class representing a pull request on github."""
+
+    def __init__(
+        self,
+        url: str,
+        number: int,
+        author: str,
+        title: str,
+        assignees: List[Assignee]
+    ) -> None:
+        self.url = url
+        self.number = number
+        self.author = author
+        self.title = title
+        self.assignees = assignees
+
+    def is_reviewer_assigned(self) -> bool:
+        """Checks whether a reviewer assigned to the PR."""
+        return not (
+            len(self.assignees) == 1 and self.assignees[0].name == self.author)
```

Comment:
This doesn't seem right. Won't it say a reviewer is assigned if there are no assignees (i.e. `len(self.assignees)` is 0)?

Reply:
Sorry, not getting your point. PR author and reviewer both counted as assignees so, if no one is assigned in a PR that means the reviewer is not assigned either

Reply:
Consider the scenario where no one is assigned to the PR. Then `len(self.assignees) == 1` evaluates to `False`, so `len(self.assignees) == 1 and self.assignees[0].name == self.author` evaluates to `False` and we return `True`. However, a reviewer is not assigned to the PR, so we should have returned `False`.

Reply:
yeah, you are right. actually where this method is being used, excludes the case where no one is assigned. Though I have updated it.